### PR TITLE
Assign Moderators by Session

### PR DESCRIPTION
**Pull request by edwardstamps**

### CODE REVIEW

File: mobile/src/home-view.js [hunk 1]
```diff
@@ -190,15 +191,16 @@ class HomeView extends PureComponent {
             this.setState({ isLoggedIn: true })
           }, 500)
         }
-        fbc.database.private.adminableUserRef('adminToken').once('value', async data => {
-          const longLivedToken = data.val()
+        fbc.database.private.adminableUserRef().once('value', async data => {
+          const longLivedToken = data.val().adminToken
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: `data.val()` may be null

File: mobile/src/table.js [hunk 2]
```diff
@@ -131,12 +134,13 @@ export default class MyList extends Component {
     </View>
   )
 
-  renderHeader = questions => {
-    const { isAdmin, showAnswer, showRecent, allQuestions } = this.props
-    if (allQuestions.length === 0 && isAdmin === false) {
+  renderHeader = (questions, isSessionAdmin) => {
+    console.log(isSessionAdmin)
```
Comment (AdamLiechty) [inline mobile/src/table.js]: remove

File: mobile/src/table.js [hunk 3]
```diff
@@ -205,9 +209,8 @@ export default class MyList extends Component {
     )
   }
 
-  renderHeaderButtons = (x, y, z) => {
-    const { isAdmin } = this.props
-    if (isAdmin)
+  renderHeaderButtons = (x, y, z, isSessionAdmin) => {
```
Comment (AdamLiechty) [inline mobile/src/table.js]: x,y, and z don't seem to be used

File: web/admin/src/Admin.js [hunk 4]
```diff
@@ -538,13 +536,16 @@ export default class Admin extends Component {
...
 
-  setAdmin(userId, isAdmin, fbc) {
-    const tokenRef = fbc.database.private.adminableUsersRef(userId).child('adminToken')
-    if (isAdmin) {
-      fbc.getLongLivedAdminToken().then(token => tokenRef.set(token))
-    } else {
-      tokenRef.remove()
-    }
+  saveModerator = (host, sessions) => {
+    console.log(sessions)
```
Comment (AdamLiechty) [inline web/admin/src/Admin.js]: remove

File: web/admin/src/Moderator.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+const Moderator = ({ moderators, openModal, selectMod, adminData }) => {
+  const [hideAdmins, setHide] = useState()
+  return (
+    <div className="containerSmall">
+      <div className="cellBoxTop">
+        <h2 className="margin-right">{t('moderators')}</h2>
+        <button className="dd-bordered" onClick={openModal}>
+          {t('add_moderator')}
+        </button>
+        <div style={{ flex: 1 }} />
```
Comment (AdamLiechty) [inline web/admin/src/Moderator.js]: don't use inline styles.

File: web/admin/src/ModeratorModal.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+          disabled={!selectedHost.length && !assignedSessions.length}
+          className="formButton"
+        >
+          {t('save')}
+        </button>
+      </div>
+    </Modal>
+  )
+
+  function returnUserData() {
```
Comment (AdamLiechty) [inline web/admin/src/ModeratorModal.js]: this looks like a copy-paste from the file above

File: mobile/src/table.js (hunk 3 above)
Comment (edwardstamps) [reply mobile/src/table.js]: x, y, z are used starting on line 235

Comment (AdamLiechty) [reply mobile/src/table.js]: ah, I missed them, sorry.  Maybe name them `popularStyle`, `recentStyle`, `answeredStyle`

File: mobile/src/home-view.js [hunk 7]
```diff
@@ -190,15 +191,16 @@ class HomeView extends PureComponent {
             this.setState({ isLoggedIn: true })
           }, 500)
         }
-        fbc.database.private.adminableUserRef('adminToken').once('value', async data => {
-          const longLivedToken = data.val()
-          if (longLivedToken) {
+        fbc.database.private.adminableUserRef().once('value', async data => {
+          //   const longLivedToken = (data.val() || {}).adminToken
+          const { adminSessions, adminToken } = data.val()
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: if `data.val()` is null, this line will fail

File: web/admin/src/ModeratorModal.js [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+          disabled={!selectedHost.length && !assignedSessions.length}
+          className="formButton"
+        >
+          {t('save')}
+        </button>
+      </div>
+    </Modal>
+  )
+
+  // function returnUserData() {
```
Comment (AdamLiechty) [inline web/admin/src/ModeratorModal.js]: remove

File: web/admin/src/ModeratorModal.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  let newArray = assignedSessions.slice()
+  if (isAssigned === -1) {
+    newArray = assignedSessions.concat([session])
+  } else {
+    newArray.splice(isAssigned, 1)
+  }
+  return (
+    <li className="modalListCell" key={session.key}>
+      <p className="cellName">{session.sessionName}</p>&nbsp;
+      <div style={{ flex: 1 }} />
```
Comment (AdamLiechty) [inline web/admin/src/ModeratorModal.js]: no inline styles

**Pull request merged**